test/gtest/uct/test_md.cc

```diff
@@ -444,8 +444,11 @@ UCS_TEST_SKIP_COND_P(test_md, alloc,
                 memset(address, 0xBB, size);
             }
 
-            status = uct_md_mkey_pack_v2(md(), mem.memh, &pack_params, &key);
-            ASSERT_UCS_OK(status);
+            if (md_attr().flags & UCT_MD_FLAG_NEED_RKEY) {
```

Comment:
why needed?
maybe some MD needs to fix its rkey packed size?

Reply:
This check is needed for cuda_copy MD.

Reply:
so what's the problem if rkey_packed_size=0?

Reply:
> why is checking rkey_size preferrable over checking cap?

According to UCT Api definition, uct_rkey_pack should work even if NEED_RKEY cap is not set.
We could also allocate 1-byte buffer in rkey_packed_size is 0 and do uct_rkey_pack to it

Reply:
But it requires additional logic on the user side.

I think it is better not to check `(mkey_buffer != NULL)` condition inside `uct_md_mkey_pack_params_check` if `rkey_packed_size==0` as it was done in this commit: https://github.com/openucx/ucx/pull/9518/commits/660070f82c87b560f520e96df1af4a7138cc320f

WDYT?

Reply:
imo, it looks weird to call function which is not supposed to be called

Reply:
> > why is checking rkey_size preferrable over checking cap?
> 
> According to UCT Api definition, uct_rkey_pack should work even if NEED_RKEY cap is not set. We could also allocate 1-byte buffer in rkey_packed_size is 0 and do uct_rkey_pack to it

where is it stated explicitly? Why would we need to pack key if it is not really needed?

Reply:
The goal of gtest in is test the APIs as much as possible, even is something is "not needed"
so we should call and test uct_rkey_pack as much as possible
here I don't see a reason to skip calling it, we can pass non-NULL pointer or 1-byte buffer if rkey_packed_size==0

Reply:
ok, but where is defined that `uct_rkey_pack` always works regardless of md caps? Seems  to be inconsistent with other caps/apis.

Reply:
@brminich @yosefe maybe then it would be better to completely remove `(mkey_buffer != NULL)` check from `uct_md_mkey_pack_params_check`? 

Then the call wouldn't fail if someone pass NULL ptr in case of `rkey_packed_size==0`, but would fail in the internal `md->ops->mkey_pack()` call if NULL was passed but `rkey_packed_size!=0`.

Reply:
imo, we should not allow to pass null buffers to the API which is expected to pack something there

Reply:
agree with @brminich to keep the NULL check